File: algorithms/2078.two-furthest-houses-with-different-colors/solution.go
func maxDistance(colors []int) int {
	l := len(colors)

	left := 0
	for colors[left] == colors[l-1] {
		left++
	}

	right := l - 1
	for colors[0] == colors[right] {
		right--
	}

	if (l - 1 - left) < right {
		return right
	} else {
		return l - 1 - left
	}
}

File: algorithms/2078.two-furthest-houses-with-different-colors/solution.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int maxDistance(vector<int>& colors) {
        int len = colors.size();

        int left = 0;
        while (colors[left] == colors[len - 1]) {
            left++;
        }

        int right = len - 1;
        while (colors[0] == colors[right]) {
            right--;
        }

        return max((len - 1 - left), right);
    }
};
